File: CarePackage/Software/Spotify.cs
namespace CarePackage.Software;

public class Spotify : BaseSoftware
{
    public override string           Key            { get; set; } = "spotify";
    public override string           Name           { get; set; } = "Spotify";
    public override SoftwareCategory Category       { get; set; } = SoftwareCategory.Media;
    public override string           DownloadName   { get; set; } = "SpotifySetup.exe";
    public override bool             IsArchive      { get; set; } = false;
    public override bool             ShouldCacheUrl { get; set; } = false;
    public override bool             RequiresAdmin  { get; set; } = false;
    public override Bitmap           Icon           { get; set; } = Resources.Icons.spotify;
    public override string           Homepage       { get; set; } = "https://spotify.com/";

    public override Task<string> GetDownloadUrlAsync(CancellationToken ct)
        => Task.FromResult("https://download.scdn.co/SpotifySetup.exe");
}

File: packages/app/src/services/software/definitions/spotify.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'spotify';
	public name = 'Spotify';
	public category = [SoftwareCategory.Audio, SoftwareCategory.Media];
	public downloadName = 'SpotifySetup.exe';
	public icon = 'spotify.png';
	public homepage = 'https://spotify.com';

	public async resolveDownloadUrl() {
		return ok('https://download.scdn.co/SpotifySetup.exe');
	}
}
